OpenRA.Mods.Common/Traits/CaptureManager.cs

```diff
@@ -221,6 +221,10 @@ public bool StartCapture(Actor self, Actor target, CaptureManager targetManager,
 			if (captures == null)
 				return false;
 
+			var enterMobile = target.TraitOrDefault<Mobile>();
```

Comment:
Can we please preface this with
```csharp
// Make sure the target has stopped moving
// HACK: We can't trust that CurrentMovementTypes is up to date, so check the internal state directly
```
?

This makes it clear to future readers why we are not using the obvious check here, which will give confidence that it can be replaced after `CurrentMovementTypes` is fixed without risking unexpected regressions.

Reply:
I still have fairly strong reservations that this check belongs here at all. This check only really makes sense when paired with a specific condition to disable `Mobile` via the mod yaml, and is motivated by the specific RA and TS mod balance where the thief has a comparable or lower speed to its targets. This may not be true in other mods.

Reply:
The alternative is to set the condition earlier via `Enter.TickInner` and triggered when within a certain distance to the target. Then we don't need any additional checks at all, but we would need to have a separate `BeforeCaptureCondition` in addition to the `CapturingCondition` just for this use case (although it isn't completely ruled out that some modder might find an additional use for it).

Reply:
From IRC discussion: This is more palatable when considered from a "is the unit aligned to the cell grid" rather than "is the unit moving" perspective.